include/minisketch.h

```diff
@@ -1,14 +1,22 @@
 #ifndef _MINISKETCH_H_
 #define _MINISKETCH_H_ 1
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include <stdint.h>
 #include <stdlib.h>
 #include <unistd.h>
 
+#ifdef __cplusplus
+#  if __cplusplus >= 201103L
```

Comment:
I've never seen preprocessor directives nested like this (and didn't realise that whitespace was permitted between the `#` and the directive). This makes it so much clearer to see the scope of the `#if`s. Really nice :+1:

Reply:
We do have a few examples in the Bitcoin Core codebase, try:

```sh
git grep '^# ' $(find -name '*.[ch]' -o -name '*.cpp')
```